### PR TITLE
Fix serialization of closed archetype slot with occurrences still present

### PR DESCRIPTION
**Pull request by pieterbos**
serialization serialized occurrences first, then closed. Should be the other way around.

The question arises whethger closed slots should have occurrences, but hey, if it's in the AOM and possible in the ADL, safe to serialize at least, and perhaps the OPT generator should not do this?

### CODE REVIEW

**Comment by pieterbos** [comment]
@J3173 I changed it so it now matches the specification. I think that is very possible. While it technically is a breaking change - none of the ADL code generated with Archie that used this 'feature' serialized correctly. So the only place where this can break if someone hand-edited a closed slot with occurrences.  That should make migration easy. Because that is unlikely to happen. Perhaps we could test with the archetypes in our systems to be sure?

**Comment by J3173** [comment]
> Perhaps we could test with the archetypes in our systems to be sure?

I have tested this change with our code and the our existing Nedap archetypes and all seems fine.

File: tools/src/test/resources/adl2-tests/features/aom_structures/slots/openEHR-EHR-SECTION.slot_closed.v1.0.0.adls [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	original_author = <
+		["name"] = <"Thomas Beale">
+ 		["email"] = <"thomas.beale@openEHR.org">
+		["organisation"] = <"openEHR Foundation <http://www.openEHR.org>">
+		["date"] = <"2010-03-14">
+	>
+	details = <
+		["en"] = <
+			language = <[ISO_639-1::en]>
+			purpose = <"Closed archetype slot with occurrences">
```
Comment (J3173) [inline tools/src/test/resources/adl2-tests/features/aom_structures/slots/openEHR-EHR-SECTION.slot_closed.v1.0.0.adls:20]: Can you update this description as the slot no longer has occurrences?

**Review by J3173 — CHANGES_REQUESTED** [review]
Looks good! One small remaining thing

File: tools/src/test/resources/adl2-tests/features/aom_structures/slots/openEHR-EHR-SECTION.slot_closed.v1.0.0.adls (hunk 1 above)
Comment (pieterbos) [reply tools/src/test/resources/adl2-tests/features/aom_structures/slots/openEHR-EHR-SECTION.slot_closed.v1.0.0.adls]: Done, could you review again? :)

**Review by J3173 — APPROVED** [review]

**Pull request merged**